src/figuresClasses.ts

```diff
@@ -1,11 +1,120 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  lue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+abstract class FigureBase {
+  static checkWrongSidesOfFigure(figureSidesArray: number[] | number): void {
+    if (typeof figureSidesArray === 'number') {
+      if (figureSidesArray <= 0) {
+        throw new Error('Radius is wrong (radius is <= 0)');
+      }
+    }
+
+    // we have to do Array.isArray(figureSidesArray),
+    // otherwise figureSidesArray.some((side: number) => side <= 0) does't work
+    if (
+      Array.isArray(figureSidesArray) &&
+      figureSidesArray.some((side: number) => side <= 0)
```

Comment:
- Remove all comments
- And this function has a few misconceptions. It shouldn't be related to the `FigureBase` as it doesn't know anything about sides, for example, the circle doesn't have any sides. The next moment you accept `figureSidesArray` but for some reason, an array could be just a number...

Reply:
Of course, geometry really impact on this code anyway, okay. Let's look at some facts:

The side of a figure is just a border of a figure, and a figure has as many sides as many angles between these sides figure has in its closed contour. So a circle has only 1 side in its contour and between 2 nearest points (between which theoreticaly lays other infinite quantity of points), laying on circle contour, angle = 0.

It was just only theory of 2D math (only mathematical model of circle, and mathematcal model !== real model), now let's consider it within our reality 😃 
In reality (and in whole world) the circle I described above (theoretical ideal circle) doesn't exist, and including creating accuracy of our technologies and sizes of atoms, the most accurate visible (without any gear) circle as possible will be a right polygon with more than 65000 angles and sides (and an angle between 2 nearest sides still tends to 0, but is not equal). I hope it was clear to understand something below...

**So what I wanted to say... Don't complicate** something due to some topic that is not related (and doesn't impact on) to a task (like geometry), really) 'cause if you delve into every topic not directly related to a task, you'll find the 'misconceptions' everywhere.

**And here are my comments for your comments:**
- FigureBase created only for comfort and shortness: it's really better to create 1 abstrct class with 1 static method inside, than write this method in every class... 'cause what if I have hundreds of figure? 
- In addition to "It shouldn't be related to the FigureBase as it doesn't know anything about sides" - it doesn't know about sides due to restriction of task rules (every class of figure must has requirable parametrs - corresponding number of sides to each figure), of course I could create a contructor in FigureBase with unlimited list of sides as a parameter and then operate with him inside every class... but I'm restricted by rules of task : )
- And sometimes I leave comments for clarity in code (like if I think other people may catch misunderstanding, reading my code, they can read why I did what I did).

About "The next moment you accept figureSidesArray but for some reason, an array could be just a number...
" - changed, but now it's not such flexible and comfortable in using as it was.